crates/goose-cli/src/commands/session.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::Result;
+
+pub fn handle_session(verbose: bool, format: String) -> Result<()> {
+    let sessions = match goose::session::list_sessions() {
+        Ok(sessions) => sessions,
+        Err(e) => {
+            tracing::error!("Failed to list sessions: {:?}", e);
+            return Err(anyhow::anyhow!("Failed to list sessions"));
+        }
+    };
+
+    match format.as_str() {
+        "json" => {
+            println!("{}", serde_json::to_string(&sessions)?);
+        }
+        _ => {
+            if sessions.is_empty() {
+                println!("No sessions found");
+                return Ok(());
+            } else {
+                println!("Available sessions:");
+                for (id, path) in sessions {
+                    if verbose {
+                        println!("  {} ({})", id, path.display());
+                    } else {
+                        println!("  {}", id);
+                    }
+                }
```

Comment:
currently the output looks like this:
```
❯ goose session list
Available sessions:
  20250307_121402
  20250307_122909
  20250307_125332
  20250307_125441
  20250307_112710
  20250307_121343
  20250310_112933
  20250307_112732
  20250307_124641
```

i think it'd be nicer to do sth similar to what we have in goose-server routes: https://github.com/block/goose/blob/fa9bb9ebb35ae2feec468caec54d55eb36d9fea0/crates/goose-server/src/routes/session.rs#L47-L79 

then we can print sth like this:
```
[session_name] - [description]  - [modified date]
```